src/ser.rs

```diff
@@ -624,9 +621,6 @@ impl<'a, 'b> ser::SerializeMap for SerializeTable<'a, 'b> {
             SerializeTable::Table { ref mut key, .. } => {
                 key.truncate(0);
                 *key = input.serialize(StringExtractor)?;
-                if key.contains('\n') {
-                    return Err(Error::KeyNewline)
-                }
```

Comment:
Luckily, there's already logic for handling quoting keys when necessary, so all we have to do is to remove this check!